extras/parse_travis_log.py

```diff
@@ -886,11 +886,12 @@ def build_return_code(status_vera_init,
     #       It may cause false positives. Even though cppcheck is executed
     #       and all its messages can be found in the log, cppcheck messages
     #       will not cause the build to fail.
-    #       To activate cppcheck, add following line to the if statement below.
-    #       Remove also the two strings '(DEACTIVATED)' behind cppcheck in
-    #       the summary_table above.
+    #       To activate cppcheck, add following lines to the if statement
+    #       below. Remove also the two strings '(DEACTIVATED)' behind
+    #       cppcheck in the summary_table above.
     #
     #       (status_cppcheck_init is None or status_cppcheck_init) and \
+    #       (number_of_msgs_in_summary(summary_cppcheck) == 0) and \
```

Comment:
@heplesser, @jougs I go along with your arguments but disagree slightly with the conclusion. 

In my opinion, the best solution is to have no switches at all and this should be the goal. For the command-line tool they are useful but not on Travis. We have strengthened the static code analysis and put this now "in production". We fully turned on VERA, wich was not the case before, and found that CPPCHECK is very critical. We do not fully oversee the impact. I see this as a transision phase. The current, simle implementation helps to handle this without doing major code changes. 

I think for CPPCHECK we need to collect information on what the impact would be. This PR therefore activates it and we run through all the logic needed for the evaluation. Just the exit code is skipped. If we turn it fully off, it probbaly will never be turned on and we have dead code. 

I would volunteer watching the builds to get a clearer picture of the impact CPPCHECK has. Later, at some point in time, we need to make a descision whether we activate or stay away from it and remove the code. If a descission has once been made and the  code analysis runs stable also the switches are not needed any longer. Therefore, I am not sure if it is worth the work adding additional logic to pass arguments through shell scripts and logs to Python.